lib/generate_puppetfile/bin.rb

```diff
@@ -175,43 +192,62 @@ def read_puppetfile (puppetfile)
       find_executable0('puppet')
     end
 
-    # Public: Download the list of modules and their dependencies to @workspace
+    # Public: Download the list of forge modules and their dependencies to @workspace
     #
-    # module_list is an array of forge module names to be downloaded
-    def download_modules(module_list)
-      puts "\nInstalling modules. This may take a few minutes.\n" unless @options[:silent]
-      module_list.each do |name|
-        if !_download_module(name)
-          $stderr.puts "There was a problem with the module name '#{name}'."
-          $stderr.puts "  Check that module exists as you spelled it and/or your connectivity to the puppet forge."
-          exit 2
-        end
+    # forge_module_list is an array of forge module names to be downloaded
+    def download_forge_modules(forge_module_list)
+      puts "\nInstalling forge modules. This may take a few minutes.\n" unless @options[:silent]
+      forge_module_list.each do |name|
+        next if _download_forge_module(name)
+        $stderr.puts "There was a problem with the module name '#{name}'."
+        $stderr.puts '  Check that module exists as you spelled it and/or your connectivity to the puppet forge.'
+        exit 2
       end
     end
 
-    # Private: Download an individual module
+    # Private: Download an individual forge module
     #
-    # _download_module
-    def _download_module(name)
+    # _download_forge_module
+    def _download_forge_module(name)
       command  = "puppet module install #{name} "
       command += @modulepath + Silence
 
       puts "Calling '#{command}'" if @options[:debug]
       system(command)
     end
-    
+
+    # Public: Download the list of git modules to @workspace
+    #
+    # git_module_list is an array of git URI to be downloaded
+    def download_forge_modules(forge_module_list)
+      puts "\nInstalling git modules. This may take a few minutes.\n" unless @options[:silent]
+      git_module_list.each do |name|
+        next if _download_git_module(name)
+        $stderr.puts "There was a problem with the module named '#{name}'."
+        $stderr.puts '  Check that the module exists at the url listed and/or your connectivity to the git repo.'
+        exit 2
+      end
+    end
+
+    # Private: Download an individual git module
+    #
+    # _download_git_module
+    def _download_git_module(git_url)
+      Rugged::Repository.clone_at(git_url, git_url.path.partition(%r{[\w-]+\/[\w-]+})[1].split('-')[-1]) rescue Rugged::Error
+    end
+
     # Public: generate the list of modules in Puppetfile format from the @workspace
-    def generate_module_output ()
+    def generate_module_output
       module_output = `puppet module list #{@modulepath} 2>/dev/null`
 
       module_output.gsub!(/\x1B\[([0-9]{1,2}(;[0-9]{1,2})?)?[m|K]/, '') # Strips ANSI color codes
       module_output.gsub!(/^\/.*$/, '')
-      module_output.gsub!(/-/,      '/')
+      module_output.tr!('-', '/')
```

Comment:
Is this from rubocop? Seems easier mentally to me to have a bunch of gsubs, rather than a gsub + tr.